onnxruntime/core/providers/coreml/model/model.h

```diff
@@ -13,6 +13,14 @@
 #include "core/common/status.h"
 #include "core/platform/ort_mutex.h"
 
+#if defined(__APPLE__)
+#ifdef __OBJC__
```

Comment:
maybe limit the declaration of `GetMLMultiArrayCopyInfo` to `defined(__OBJC__)`? otherwise the signature will be different (I think) between normal C++ and Objective-C++ compilation. we don't use the helper outside of an Objective-C++ context.